spec/graphql/static_validation/rules/mutation_root_exists_spec.rb

```diff
@@ -0,0 +1,39 @@
+require "spec_helper"
+
+describe GraphQL::StaticValidation::MutationRootExists do
+  let(:query_string) {%|
+    mutation addBagel {
+      introduceShip(input: {shipName: "Bagel"}) {
+        clientMutationId
+        shipEdge {
+          node { name, id }
+        }
+      }
+    }
+  |}
+
+  let(:schema) {
+    query_root = GraphQL::Field.define do
+      name "Query"
+      description "Query root of the system"
+    end
+
+    GraphQL::Schema.define do
+      query query_root
+    end
+  }
+
+  let(:validator) { GraphQL::StaticValidation::Validator.new(schema: schema, rules: [GraphQL::StaticValidation::MutationRootExists]) }
+  let(:query) { GraphQL::Query.new(schema, query_string) }
+  let(:errors) { validator.validate(query)[:errors] }
+
+  it "errors when a mutation is performed on a schema without a mutation root" do
+    assert_equal(1, errors.length)
+    missing_mutation_root_error = {
+      "message"=>"Schema is not configured for mutations",
+      "locations"=>[{"line"=>2, "column"=>5}],
+      "path"=>["mutation addBagel"],
```

Comment:
It generally leads to the field that has an error, like a more human-friendly version of `locations`. To me it makes sense here: some part of your query caused an error, if you want to find it, follow this path. Does it make sense in that context? If you think `nil` is a better fit, feel free to pass it.